suite/tests/api/detach_state.c

```diff
@@ -568,12 +627,74 @@ START_FILE
         pop      r12 @N@ \
         pop      REG_XBX @N@ \
         pop      REG_XBP @N@
-#else /* X64 */
+#elif defined(X86) && !defined(X64)
+/**************************************************
+ * X86_32
+ */
 #    define PUSHALL \
         pusha
 #    define POPALL  \
         popa
 #    error NYI /* TODO i#3160: Add 32-bit support. */
+#elif defined(AARCH64)
+/**************************************************
+ * AARCH64
+ */
+#    define PUSH_GPRS                           \
+        /* SP is checked separately so value here doesn't matter. */ \
+        stp      x30, x0, [sp, #-16]! @N@\
+        stp      x28, x29, [sp, #-16]! @N@\
+        stp      x26, x27, [sp, #-16]! @N@\
+        stp      x24, x25, [sp, #-16]! @N@\
+        stp      x22, x23, [sp, #-16]! @N@\
+        stp      x20, x21, [sp, #-16]! @N@\
+        stp      x18, x19, [sp, #-16]! @N@\
+        stp      x16, x17, [sp, #-16]! @N@\
+        stp      x14, x15, [sp, #-16]! @N@\
+        stp      x12, x13, [sp, #-16]! @N@\
+        stp      x10, x11, [sp, #-16]! @N@\
+        stp      x8, x9, [sp, #-16]! @N@\
+        stp      x6, x7, [sp, #-16]! @N@\
+        stp      x4, x5, [sp, #-16]! @N@\
+        stp      x2, x3, [sp, #-16]! @N@\
+        stp      x0, x1, [sp, #-16]!
+#        define PUSHALL                         \
```

Comment:
nit: Indentation can be fixed. Or is that due to clang-format?

Reply:
Fixed, here and below.

The whole asm side of the file has clang-format off b/c it was doing the wrong thing, I think b/c it thinks it's still C code (all of the split-asm files are like that).